crates/holochain/src/conductor/space.rs

```diff
@@ -107,8 +108,12 @@ pub struct Space {
     /// A cache for slow database queries.
     pub dht_query_cache: DhtDbQueryCache,
 
-    /// Countersigning workspace that is shared across this cell.
-    pub countersigning_workspace: CountersigningWorkspace,
+    /// Countersigning workspace for session state.
+    pub countersigning_workspaces:
+        Arc<parking_lot::Mutex<HashMap<CellId, Arc<CountersigningWorkspace>>>>,
```

Comment:
@ThetaSinner For a cell there can only be 1 countersigning session. The `CountersigningWorkspaceInner` has an `Option<CountersigningSessionState>`. Isn't that a redundant optionality now? If there's no countersigning session for a cell, it won't have a workspace and won't be in the hash map `countersigning_workspaces`. Is that correct?

Reply:
The workspace exists from when the cell is created. The workspace either has a session or it doesn't. I don't want to deal with building a workspace when a session starts because there are several entry and exit points to the workflow.

Reply:
Okay, I adapted to it.